py-polars/src/dataframe.rs

```diff
@@ -425,13 +425,33 @@ impl PyDataFrame {
         sep: u8,
         quote: u8,
         batch_size: usize,
-        datetime_format: Option<String>,
+        mut datetime_format: Option<String>,
         date_format: Option<String>,
         time_format: Option<String>,
         float_precision: Option<usize>,
         null_value: Option<String>,
     ) -> PyResult<()> {
         let null = null_value.unwrap_or(String::new());
+
+        // if datetime format not specified, infer the maximum required precision
```

Comment:
Can we do this in `polars-io`? Then the rust libraries behavior will match that of the python bindings.

Reply:
Ahh, good pointer; that does seem a better home for it :) 
Have updated accordingly (just needed to take a mutable ref to SerializeOptions so I could update It post-inference).